### DISCUSSION TITLE
Elgato Stream Deck Intergration

### DISCUSSION THREAD

**Discussion by bubaphex** [kimai/kimai#4371] [Plugins]
Has anyone explored integration with an Elgato Stream Deck? [Toggl](https://github.com/tobimori/streamdeck-toggl) has a pretty decent one. Google has come up short on options for Kimai.

Any interest from the community?

**Comment by kevinpapst** [comment]
I would love to test that. 
Have it on my Amazon wishlist since forever, but nobody wanted to sponsor it yet 🤷 so I have to watch it from the distance for now.
But I am super curious anyway and the linked code looks pretty simple, so should be fairly easy to get it up and running.

**Comment by b263** [comment]
I have an early version of a Stream Deck plugin available here: https://github.com/b263/stream-deck-time-tracker

It currently has an action for starting/stoping tracking and displays the time tracked today and in the current period. An action for simple time reports will be added later.

Looking forward to some feedback.

**Reply by bubaphex** [reply]
Thanks @b263 I've just downloaded it and will give it a test run!

**Reply by pdmarf** [reply]
very interested in this. I'm trying out the @b263 plugin and it seems to work ok. Is it still maintained?

**Reply by b263** [reply]
@pdmarf I just updated it to use the new auth method. But I can't add new features currently.

**Reply by kevinpapst** [reply]
Thanks @b263 👍

**Comment by ignaz-pichler** [comment]
Any Update in 2026? 
@b263 the plugin is not online anymore. Does it still work?

**Reply by markbking** [reply]
This is/was a great plugin for Stream Deck interface with Kimai. I hope the developer considers making it available again! Or at least the code for someone else to launch from. @b263

**Comment by andrewmitchell-automations** [comment]
Hey all! Late to this thread, but for anyone still looking: I built a new Kimai StreamDeck plugin. Free, MIT licensed.

https://github.com/alphaomegateam/streamdeck-kimai/releases/tag/v0.1.0

This is a separate codebase, not a fork or continuation of @b263's. It's built against the current API.

Four key types: start a timer for a project and activity you choose, stop whatever is running, a display key showing the current client and elapsed time that updates live, and a toggle key that starts its own timer or stops it if it's already the one running. Configuration is all in the Stream Deck app. You enter your Kimai URL and an API token once, and the project and activity dropdowns fill in from your own instance.

What I need most: **Windows testers.** I've only run this on macOS, against 2.57.0. The manifest declares Windows 10+ but I have never once launched it there. If you use Kimai on Windows and have five minutes, that's the gap I'd most like closed.

Two API questions, while I have the attention of people who would know (cc @kevinpapst):

1. On `/api/activities`, I found `globals=1` returns *only* global activities and overrides the `project` filter, so a project's own activities end up hidden. Dropping the parameter and sending just `?project={id}&visible=1` gives me what I actually wanted. Is that the intended behavior of the flag?

2. With `Active Entries Limit: 1`, does a start request arriving while another timer runs auto-stop the first, or get rejected? My "stop the running timer first" option defaults to on, so the default path never hits it, but I'd like the off path to behave correctly rather than guess.

Needs the Stream Deck app 7.1 or newer. Feel free to send thoughts, bugs, or new button requests here or via an issue on that repo.

_Note: I am a developer, but I did use Claude Code to create the plugin, so if you're philosophically opposed to using software which AI had a hand in... I don't know what to tell ya, it's 2026 🤷🏻‍♂️_

**Reply by kevinpapst** [reply]
Awesome job, I am going to order a Stream Deck to test that integration.
Can you recommend one? 
It seems the main options are:
- Elgato Stream Deck Neo
- Elgato Stream Deck Plus
Which one do you use?

**Reply by markbking** [reply]
andrewmitchell-automations:  You are THE MAN!. I am now running it on Windows 11 with a Stream Deck XL and self hosted Kimai site on an Unraid Docker for time tracking for multiple projects. I will happily report any issues should they occur.

**Reply by andrewmitchell-automations** [reply]
Thanks all!! @kevinpapst I have an OLD streamdeck - the very first one to come out. [This](https://www.elgato.com/us/en/p/stream-deck?gad_campaignid=23619789356&gbraid=0AAAAA9ciIxVhtPgeiLTNKYkK6FoCrGSok) is a slightly newer version of the same 5x3 button layout I use, but the plugin should work across any device.

**Reply by andrewmitchell-automations** [reply]
@markbking thank you for testing! Feel free to report any issues or feature requests!

**Reply by markbking** [reply]
@andrewmitchell-automations   So far everything is working extremely well. The fact that the "Current Timer" and "Stop Timer" functions work with multiple iterations of the buttons at the same time is brilliant. I have a Time page setup with all of my open projects that I start tracking with and then keep multiple iterations of the Current Timer and Stop Timer buttons on other pages so that I can monitor and stop time tracking from those pages instead of going back to my Time page. Awesome implementation!

I would love to help you share the availability and review of this plugin but am not sure where to do so. If you have any thoughts, let me know.

**Reply by andrewmitchell-automations** [reply]
@markbking glad you're finding it useful! I have posted in r/Kimai, and submitted the plugin to the Elgato plugin marketplace for approval (first try denied - I didn't provide enough info!) but I'm open to any other ideas you may have for how to get visibility on it! @kevinpapst may also have some ideas here.
